grpc/src/main/java/io/stargate/grpc/impl/EventNotifier.java

```diff
@@ -0,0 +1,116 @@
+package io.stargate.grpc.impl;
+
+import io.grpc.netty.shaded.io.grpc.netty.CustomChannelFactory;
+import io.stargate.db.EventListenerWithChannelFilter;
+import java.net.InetAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+public class EventNotifier implements EventListenerWithChannelFilter {
+
+  private final CustomChannelFactory customChannelFactory;
+
+  public EventNotifier(CustomChannelFactory customChannelFactory) {
+    this.customChannelFactory = customChannelFactory;
+  }
+
+  @Override
+  public void onClose(Predicate<Map<String, String>> headerFilter) {
+    if (headerFilter != null) {
+      customChannelFactory.closeFilter(headerFilter);
+    }
+  }
+
+  @Override
+  public void onCreateKeyspace(String keyspace, Predicate<Map<String, String>> headerFilter) {}
```

Comment:
would be nice to have `EventListenerWithChannelFilterAdapter` and then you don't have to implement all methods from the `EventListenerWithChannelFilter`?